File: Code Chef/JavaScript/700-799/750 Far from origin FARFROMO.js
let input = '';
process.stdin.on('data', (data) => input += data);

const strToInt = (strArray) => {
    strArray.shift();
    const intArray = strArray.map((strItem) => strItem.trim().split(' ').map(Number));
    
    return intArray;
};

const getDistance = (xCoordinate, yCoordinate) => {
    const sum = Math.pow(xCoordinate, 2) + Math.pow(yCoordinate, 2);
    const distance = Math.sqrt(sum);
    
    return distance;
};

const whoIsFarFromChef = (x1OfAlex, y1OfAlex, x2OfBob, y2OfBob) => {
    const distanceOfAlex = getDistance(x1OfAlex, y1OfAlex);
    const distanceOfBob = getDistance(x2OfBob, y2OfBob);
    
    (distanceOfAlex === distanceOfBob)
        ? console.log('EQUAL')
        : (distanceOfAlex > distanceOfBob)
            ? console.log('Alex')
            : console.log('BOB');
};

process.stdin.on('end', () => {
    const testcases = strToInt(input.trim().split('\n'));
    
    testcases.forEach((testcase) => {
        const [x1OfAlex, y1OfAlex, x2OfBob, y2OfBob] = testcase;
        whoIsFarFromChef(x1OfAlex, y1OfAlex, x2OfBob, y2OfBob);
    });
});

File: Code Chef/C Programming/700-799/750 Far from origin FARFROMO.c
#include <stdio.h>
#include <math.h>
int main(void)
{
	int t,x1,x2,y1,y2;
	scanf("%d",&t);
	for(int i=0;i<t;i++)
	{
	    scanf("%d %d %d %d",&x1,&y1,&x2,&y2);
	    if((sqrt(pow(x1,2)+pow(y1,2)))==(sqrt(pow(x2,2)+pow(y2,2))))
	    {
	        printf("EQUAL\n");
	    }
	    else if((sqrt(pow(x1,2)+pow(y1,2)))<(sqrt(pow(x2,2)+pow(y2,2))))
	    {
	        printf("BOB\n");
	    }
	    else if((sqrt(pow(x1,2)+pow(y1,2)))>(sqrt(pow(x2,2)+pow(y2,2))))
	    {
	        printf("ALEX\n");
	    }
	}
	return 0;
}